src/components/Resource/Forms/Training/TrainingImages.tsx

```diff
@@ -0,0 +1,507 @@
+import {
+  ActionIcon,
+  Button,
+  Center,
+  Checkbox,
+  createStyles,
+  Group,
+  Image,
+  Loader,
+  Pagination,
+  Paper,
+  SimpleGrid,
+  Stack,
+  Text,
+  Textarea,
+  Title,
+  useMantineTheme,
+} from '@mantine/core';
+import { FileWithPath } from '@mantine/dropzone';
+import { showNotification, updateNotification } from '@mantine/notifications';
+import { ModelFileVisibility } from '@prisma/client';
+import { IconCheck, IconTrash, IconX } from '@tabler/icons-react'; // import { saveAs } from 'file-saver';
+import JSZip from 'jszip';
+import { isEqual } from 'lodash-es';
+import { useEffect, useState } from 'react';
+import { ImageDropzone } from '~/components/Image/ImageDropzone/ImageDropzone';
+import { goBack, goNext } from '~/components/Resource/Forms/Training/TrainingCommon';
+import { UploadType } from '~/server/common/enums';
+import { IMAGE_MIME_TYPE } from '~/server/common/mime-types';
+import { createModelFileDownloadUrl } from '~/server/common/model-helpers';
+import { useS3UploadStore } from '~/store/s3-upload.store';
+import { ModelById } from '~/types/router';
+import { showErrorNotification, showSuccessNotification } from '~/utils/notifications';
+import { bytesToKB } from '~/utils/number-helpers';
+import { trpc } from '~/utils/trpc';
+
+// zustand
+
+interface imageDataType {
+  name: string;
+  type: string;
+  url: string;
+  caption: string;
+}
+
+const useStyles = createStyles(() => ({
+  imgOverlay: {
+    position: 'relative',
+    '&:hover .trashIcon': {
+      display: 'initial',
+    },
+  },
+  trash: {
+    display: 'none',
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    zIndex: 10,
+    margin: 4,
+  },
+}));
+
+// TODO [bw] is this enough? do we want jfif and webp?
+const imageExts: { [key: string]: string } = {
+  png: 'image/png',
+  jpg: 'image/jpeg',
+  jpeg: 'image/jpeg',
+  // webp: 'image/webp',
+};
+
+export const TrainingFormImages = ({ model }: { model: ModelById }) => {
+  const [imageList, setImageList] = useState<imageDataType[]>([]);
+  const [initialImageList, setInitialImageList] = useState<imageDataType[]>([]);
+  const [page, setPage] = useState(1);
+  const [ownRights, setOwnRights] = useState<boolean>(false);
+  const [shareDataset, setShareDataset] = useState<boolean>(false);
+  const [zipping, setZipping] = useState<boolean>(false);
+  const [loadingZip, setLoadingZip] = useState<boolean>(false);
+  const theme = useMantineTheme();
+  const { classes, cx } = useStyles();
+  const queryUtils = trpc.useContext();
+  const { upload, getStatus: getUploadStatus } = useS3UploadStore();
```

Comment:
https://www.dropbox.com/scl/fi/xfljes567xiot9ul2db3i/2023-08-24-13-47-12.mp4?rlkey=hsqjd4lwrk9xtqfzqinhzb4bq&dl=0

Reply:
thanks for the detailed vid - the zustand stuff is in the known issues section, and at this point I've just kind of run out of time for it - let's consider this a fast follow for now.